enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/hooks/use-sdk-iframe-embed-event-bus.ts

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState } from "react";
+import { match } from "ts-pattern";
+
+import { isWithinIframe } from "metabase/lib/dom";
+
+import type {
+  SdkIframeEmbedMessage,
+  SdkIframeEmbedSettings,
+} from "../types/embed";
+
+type Handler = (event: MessageEvent<SdkIframeEmbedMessage>) => void;
+
+export function useSdkIframeEmbedEventBus(): {
+  embedSettings: SdkIframeEmbedSettings | null;
+} {
+  const [embedSettings, setEmbedSettings] =
+    useState<SdkIframeEmbedSettings | null>(null);
+
+  useEffect(() => {
+    const messageHandler: Handler = (event) => {
+      if (!isWithinIframe() || !event.data) {
+        return;
+      }
+
+      match(event.data).with(
+        { type: "metabase.embed.setSettings" },
+        ({ data }) => setEmbedSettings(data),
+      );
+    };
+
+    window.addEventListener("message", messageHandler);
+
+    // notify embed.js that the iframe is ready
+    window.parent.postMessage({ type: "metabase.embed.iframeReady" }, "*");
+
+    return () => {
+      window.removeEventListener("message", messageHandler);
+    };
+  });
```

Comment:
Is it intentional that we run this effect every render? I suppose we only do it once.

```suggestion
  }, []);
```

Reply:
Good catch, can't believe I missed that.

Reply:
Fixed in https://github.com/metabase/metabase/pull/56915/commits/3099b7d158898d4b84c5efc444483847494e73ee, thanks!